app/models/order_cycle.rb

```diff
@@ -250,6 +250,13 @@ def refresh_products_cache
     OpenFoodNetwork::ProductsCache.order_cycle_changed self
   end
 
+  def items_bought_by_user(user, distributor)
+    # The Spree::Order.complete scope only checks for completed_at date, does not ensure state is "complete"
+    orders = Spree::Order.complete.where(state: "complete", user_id: user, distributor_id: distributor, order_cycle_id: self)
+    items = orders.map(&:line_items).flatten
```

Comment:
Here we're introducing a `N+1` query, right? ActiveRecord can help you to avoid it, instead of using `#map()`.